app/controllers/github/workflows_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Github
+  # Handles displaying workflow info for an app
+  class WorkflowsController < BaseController
+    before_action :set_github_workflow, only: %i[show]
+    before_action :set_github_workflow_runs, only: %i[index]
+
+    # GET /github/workflows or /github/workflows.json
+    def index
+      if request.path.include?('deploy')
+        deploy_workflow = @github_repository.deploy_workflow
+        redirect_to team_app_deploy_path(@team, @app, deploy_workflow.id) if deploy_workflow
+      else
+        @current_page = params.fetch(:page, 1)
+        @github_workflows = @github_repository.workflows[:workflows]
+        set_pages
+      end
+    end
+
+    # GET /github/workflows/1 or /github/workflows/1.json
+    def show
+      @current_page = params.fetch(:page, 1)
+      @all_workflows = if request.path.include?('deploy')
+                         [@github_repository.deploy_workflow]
+                       else
+                         @github_repository.workflows[:workflows]
+                       end
+      @github_workflow_runs = @github_workflow.workflow_runs(params[:page] || 1, { branch: params[:ref] }).to_h
+      set_pages
+    end
+
+    private
+
+    # Use callbacks to share common setup or constraints between actions.
```

Comment:
I wonder if some of these private methods could be moved into a helper to dry up the controller?

Reply:
Yeah I could move some to the base controller.